en/developer_guide/block_parameter.rst

```diff
@@ -0,0 +1,126 @@
+Block parameter
+===============
+
+Context
+-------
+
+To be properly displayed on the website, some blocks will require environment parameters.
+Those parameters could be of multiple types, for instance :
+
+ - url parameters
+ - server parameters
+ - header parameters
+ - ...
+
+An easy way to give those parameters to the block would have been to give all the request
+parameters when the block is called.
+
+This easy solution is not optimal for a main reason: when loading blocks via `ESI`_, each
+block will be cached for each url variant.
+
+If the block 1 does not require any parameters, in the following exemple :
```

Comment:
There's no end to this sentence : "if **** " should call a "then **** "
The previous sentence could be completed like this to introduce the example :

    [...] will be cached for each url variant, which is a failure if the cache produced are the same. For instance if th block 1 does not require any parameter, the following requests will produce two different caches for the same result: [...]